File: src/TmrOverlay.App/Overlays/Status/StatusOverlayDefinition.cs
using TmrOverlay.Core.Overlays;

namespace TmrOverlay.App.Overlays.Status;

internal static class StatusOverlayDefinition
{
    public static OverlayDefinition Definition { get; } = new(
        Id: "status",
        DisplayName: "Collector Status",
        DefaultWidth: 520,
        DefaultHeight: 176,
        Options:
        [
            OverlaySettingsOptionDescriptor.Boolean(
                OverlayOptionKeys.StatusCaptureDetails,
                "Show capture path",
                defaultValue: true),
            OverlaySettingsOptionDescriptor.Boolean(
                OverlayOptionKeys.StatusHealthDetails,
                "Show health details",
                defaultValue: true)
        ]);
}

File: local-mac/TmrOverlayMac/Sources/TmrOverlayMac/Overlays/Status/StatusOverlayDefinition.swift
import AppKit

enum StatusOverlayDefinition {
    static let definition = OverlayDefinition(
        id: "status",
        displayName: "Collector Status",
        defaultSize: NSSize(width: 520, height: 150)
    )

    static var defaultSize: NSSize {
        definition.defaultSize
    }
}
